### ISSUE TITLE
How to set user defined CSS?

### ISSUE THREAD

**Issue by francwalter** [marktext/marktext#3712]
### Question description

How can I set, maybe in preferences, CSS for the different elements?

I switched for my personal notes to MarkDown, which is great, I use this Editor, which is nice.
But I have difficulties to see the difference between e.g. h2 and h3 headings.
I would like to give it maybe bigger font-size difference or maybe a color.

That would only be for my own view, thats enough for me, but is that even possible?

Thanks, frank

**Comment by DoctorSubtilis** [comment]
This feature would be very important and useful!

**Comment by liushengxin** [comment]
> But I have difficulties to see the difference between e.g. h2 and h3 headings.

Yes, I need that feature , I also can not distinguish h3 , h2 ,especially in Chinese text.

As you can see bellow, 

![](https://raw.githubusercontent.com/liushengxin/ImageHosting/main/MarkText_vBxKgXGdp0.png)

**Comment by Jocs** [comment]
MarkText supports custom CSS via the **`customCss`** field in **Preferences > Theme**. Add any CSS rules there to style editor elements:
```css
body { font-family: 'My Font', sans-serif; }
h1 { color: #2563eb; }
```
Closing as answered.

**Comment by francwalter** [comment]
> MarkText supports custom CSS via the **`customCss`** field in **Preferences > Theme**. Add any CSS rules there to style editor elements:
> 
> body { font-family: 'My Font', sans-serif; }
> h1 { color: #2563eb; }

Thanks, but I dont find that customCSS field in my MarkText (v0.17.1 on Win11) Preferences > Theme:

<img width="950" height="650" alt="Image" src="https://github.com/user-attachments/assets/40b7d12e-0278-4722-9899-203a9edeae4b" />

**Comment by Jocs** [comment]
> > MarkText supports custom CSS via the **`customCss`** field in **Preferences > Theme**. Add any CSS rules there to style editor elements:
> > body { font-family: 'My Font', sans-serif; }
> > h1 { color: #2563eb; }
> 
> Thanks, but I dont find that customCSS field in my MarkText (v0.17.1 on Win11) Preferences > Theme:
> 
> <img alt="Image" width="950" height="650" src="https://private-user-images.githubusercontent.com/927269/594075211-40b7d12e-0278-4722-9899-203a9edeae4b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzkxNDc0MjksIm5iZiI6MTc3OTE0NzEyOSwicGF0aCI6Ii85MjcyNjkvNTk0MDc1MjExLTQwYjdkMTJlLTAyNzgtNDcyMi05ODk5LTIwM2E5ZWRlYWU0Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUxOFQyMzMyMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNzFlMjQ5ZGZiODVkODAxNThmYjQxZjg3NWY2MzA3YWM5NDA5OGQ5NTNmYWJmYzc3MDE2YThmNDRhOGM0Mzg4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.9coXN4oLr_SQT7D1DXBQEddQypjMUL6HyvXRqP5OaEA">

On the develop branch.

**Comment by francwalter** [comment]
> ... 
> On the develop branch.

Is there a download for that branch? 
I haven't compiled MarkText yet myself and I only find 0.17.1 downloads.
Thanks!

**Comment by francwalter** [comment]
After searching here in the repo issues, I notice that there is no HowTo for compiling it on a Windows machine.
... (deleted for better knowing now) ...

EDIT: after searching more, I found this: https://github.com/marktext/marktext/issues/4191 and all will be good 😄 
I immeditately installed the release from https://github.com/Tkaixiang/marktext and there it is:

<img width="963" height="805" alt="Image" src="https://github.com/user-attachments/assets/18221270-1344-43e0-9402-13daa8b1fe3b" />

Very very good! Thanks a lot! Also for coming back!

**Comment by francwalter** [comment]
My header CSS is now (not the last word, but much much better):

```
h1, h2, h3, h4, h5, h6 {
  line-height: 1.3;
  margin-top: 1em;
  margin-bottom: 0.5em;
}
h1, h2 {
  font-size: 2.8rem;  
}
h1 {
  font-weight: 800;   
}
h2 {
  font-weight: 400;   
}
h3, h4 {
  font-size: 1.8rem;   
}
h3 {
  font-weight: 700;   
}
h4 {
  font-weight: 400;   
}
h5, h6 {
  font-size: 1.25rem; 
}
h5 {
  font-weight: 700;   
}
h6 {
  font-weight: 400;   
}
h1 { color: #e63946; } /* red */
h2 { color: #f4a261; } /* orange */
h3 { color: #e9c46a; } /* yellow */
h4 { color: #6a994e; } /* green */
h5 { color: #2a9d8f; } /* turquois */
h6 { color: #4483C3; } /* blue */
```
Thats with several sizes and several colors, much better to distinguish.
